src/views/TransactionsCategories/AddCategory/AddCategory.js

```diff
@@ -0,0 +1,57 @@
+import React, { useState } from "react";
+import { AddCategoryForm } from "../AddCategoryForm/AddCategoryForm";
+import Modal from "../../../components/Modal";
+import { useModal } from "../../../services/hooks/useModal";
+
+export const AddCategory = ({ createCategory }) => {
+  const { isVisible, toggleVisibility } = useModal();
+  const [name, setName] = useState("");
+  const [icon, setIcon] = useState("");
+  const [color, setColor] = useState("");
+
+  const handleName = (name) => {
+    setName(name);
+  };
+
+  const handleColor = (color) => {
+    setColor(color);
+  };
+
+  const handleIcon = (icon) => {
+    setIcon(icon);
+  };
+
+  const handleCreateCategory = () => {
+    if (name === "" || icon === "") {
+      return;
+    } else {
+      createCategory({
+        id: Math.floor(Math.random() * 100),
```

Comment:
why 100? and what is this value exactly ?

Reply:
I could be wrong but it supposed to be value from range 1-99. Anyway i will use uuid library.